File: frontend/demo/component/markdown/markdown-line-breaks.ts
import '@vaadin/markdown';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('markdown-line-breaks')
export class MarkdownLineBreaks extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    // tag::snippet[]
    const markdownText = `
Hi Maria,
Your order shipped this morning.
It should arrive by Thursday.

Best regards,
Tom
    `;

    return html`<vaadin-markdown line-breaks .content=${markdownText}></vaadin-markdown>`;
    // end::snippet[]
  }
}

File: src/main/java/com/vaadin/demo/component/markdown/MarkdownLineBreaks.java
package com.vaadin.demo.component.markdown;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.markdown.Markdown;
import com.vaadin.flow.router.Route;

@Route("markdown-line-breaks")
public class MarkdownLineBreaks extends Div {

    public MarkdownLineBreaks() {
        // tag::snippet[]
        String markdownText = """
                Hi Maria,
                Your order shipped this morning.
                It should arrive by Thursday.

                Best regards,
                Tom
                """;

        Markdown markdown = new Markdown(markdownText);
        markdown.setLineBreaks(true);
        // end::snippet[]
        add(markdown);
    }

    public static class Exporter extends DemoExporter<MarkdownLineBreaks> { // hidden-source-line
    } // hidden-source-line
}
